flag-app/flag-app-project/src/components/CardDetail.tsx

```diff
@@ -0,0 +1,78 @@
+import { useDarkMode } from "../hooks";
+import { BorderCountry } from "./BorderCountry";
+import CardItem from "./CardItem";
+import { Country } from "../interfaces";
+
+import { FaArrowLeft } from "react-icons/fa6";
+
+interface Props {
+  handleClickBack: () => void;
+  country: Country;
+}
+
+export const CardDetail = ({ handleClickBack, country }: Props) => {
+  const { mode } = useDarkMode();
+
+  const { name, population, region, capital, tld, currencies, languages } =
+    country;
+
+  const countryNameCode = Object.keys(name.nativeName)[0];
+  const nativeName = name.nativeName[countryNameCode].official;
+
+  const domain = tld[0];
+
+  const countryCurrencyCode = Object.keys(currencies)[0];
+  const currency = currencies[countryCurrencyCode].name;
+
+  const languageCode = Object.keys(languages)[0];
+  const language = languages[languageCode];
```

Comment:
You could try and write a custom hook, `useCountryDetails(country)`, which would return all this data from one "central" place. This adds reusability to your codebase